gips/data/landsat/landsat.py

```diff
@@ -312,11 +314,23 @@ def query_service(cls, asset, tile, date):
             )['data']
 
             for result in response['results']:
-                available.append({
-                    'basename': result['displayId'] + '.tar.gz',
-                    'sceneID': result['entityId'],
-                    'dataset': dataset,
-                })
+                metadata = requests.get(result['metadataUrl']).text
+                xml = ElementTree.fromstring(metadata)
+                scene_cloud_cover = list(
+                    xml.find(".//{http://earthexplorer.usgs.gov/eemetadata.xsd}metadataField[@name='Scene Cloud Cover']")
+                )[0].text
```

Comment:
Near as I can tell `Element.find` returns a single object so I don't think you have to `list(...)[0]`.

Reply:
So this is a strange little bit of the ElementTree API: in order to get the children of an element, you cast it to a list. This is in favor of the existing but deprecated `getchildren()` method. So I'm actually not indexing the results of `Element.find` but instead the list of it's children.

Reply:
Since that is infinitely unclear, maybe a comment explaining the cast to a list would be helpful?

Reply:
Oops.  : ]  `list(o)` isn't `[o]`, it's `[i for i in o]`, which is very different; my mistake.

So I have a different suggestion:  The idiomatic way to get the first item out of an iterable in python is `scene_cloud_cover = next(xml.find(...))`.  It's much harder to misunderstand.  : P

But a comment would do just as well, so it's up to you.

Reply:
Annoyingly, `Element` is not an iterator so `next()` doesn't work, but looking into the source I've realized I don't need the list cast, I can simply index an instance of `Element`. So I think I will do that and explain in a comment.